src/warnings.cpp

```diff
@@ -56,6 +62,11 @@ bilingual_str GetWarnings(bool verbose)
         warnings_verbose.emplace_back(warnings_concise);
     }
 
+    if (g_timeoffset_warning) {
+        warnings_concise = _("Your computer's date and time appear out of sync with the network, this may lead to consensus failure. Please ensure it is correct.");
+        warnings_verbose.emplace_back(warnings_concise);
+    }
```

Comment:
Once `g_timeoffset_warning` is set to `true` it will never be set back to `false` and this will warning will persist even after the problem is resolved?

Reply:
Correct, as is roughly the current behaviour, as per https://github.com/bitcoin/bitcoin/pull/29623#discussion_r1531234359